ui/app/common/concept-set/directives/conceptSet.js

```diff
@@ -238,7 +238,7 @@ angular.module('bahmni.common.conceptSet')
                 };
 
                 var runFormConditionForAllObsRecursively = function (formName, rootObservation) {
-                    _.each(rootObservation.groupMembers, function (observation) {
+                    rootObservation.groupMembers.forEach(function (observation) {
```

Comment:
Hi @maharjunm, this fix will fail if `rootObservation.groupMembers` is `undefined`. Afterthat `_.each` is a `null` safe operation which internally uses the `forEach` method.

Reply:
I also don't see how this change would fix anything. And there's no unit test.

Reply:
We made a commit for undefined checks for `rootObservation` and `rootObservation.groupMembers`.

`_.each` does not adhere to concurrent modification of iterating list. Using `forEach` that works for concurrent modification

Reply:
@sai789 Can you also add tests for this ?